index.js

```diff
@@ -1,500 +1,411 @@
 const BigNumber = require('bignumber.js')
-const assert = require('assert')
-const { ChannelWatcher } = require('ilp-plugin-xrp-paychan-shared')
+const StoreWrapper = require('./store-wrapper')
+const {
+  ChannelWatcher,
+  util
+} = require('ilp-plugin-xrp-paychan-shared')
 
 // constants
 const CHANNEL_KEYS = 'ilp-plugin-xrp-paychan-channel-keys'
-const DEFAULT_REFUND_THRESHOLD = 0.9
-const {
-  DEFAULT_WATCHER_INTERVAL,
-  STATE_NO_CHANNEL,
-  STATE_CREATING_CHANNEL,
-  STATE_CHANNEL,
-  POLLING_INTERVAL_OUTGOING_PAYCHAN,
-  MIN_SETTLE_DELAY,
-  xrpToDrops,
-  dropsToXrp,
-  sleep
-} = require('./src/lib/constants')
-
-// utility functions
-const randomTag = () => bignum.fromBuffer(crypto.randomBytes(4), {
-  endian: 'big',
-  size: 4
-}).toNumber()
-
-const encodeClaim = (amount, id) => Buffer.concat([
-  Buffer.from('CLM\0'),
-  Buffer.from(id, 'hex'),
-  bignum(amount).toBuffer({
-    endian: 'big',
-    size: 8
-  })
-])
-
-const computeChannelId = (src, dest, sequence) => {
-  const preimage = Buffer.concat([
-    Buffer.from('\0x', 'ascii'),
-    Buffer.from(addressCodec.decodeAccountID(src)),
-    Buffer.from(addressCodec.decodeAccountID(dest)),
-    bignum(sequence).toBuffer({ endian: 'big', size: 4 })
-  ])
-
-  return crypto.createHash('sha512')
-    .update(preimage)
-    .digest()
-    .slice(0, 32) // first half sha512
-    .toString('hex')
-    .toUpperCase()
-}
-
-const claimFunds = async (ctx) => {
-  const self = ctx.state
-
-  const bestClaim = await self.incomingClaim.getMax()
-  const signature = bestClaim.data
-  const amount = bestClaim.value
-  const claimedValue = (await self.incomingClaimSubmitted.getMax()).value
-  if (claimedValue === amount) return
-
-  const tx = await self.api.preparePaymentChannelClaim(self.address, {
-    balance: dropsToXrp(amount),
-    channel: self.incomingPaymentChannelId,
-    signature: signature.toUpperCase(),
-    publicKey: self.incomingPaymentChannel.publicKey
-  })
-
-  const signedTx = self.api.sign(tx.txJSON, self.secret)
-  ctx.plugin.debug('submitting claim transaction ', tx)
-  const {resultCode, resultMessage} = await self.api.submit(signedTx.signedTransaction)
-  if (resultCode !== 'tesSUCCESS') {
-    ctx.plugin.debug('Error submitting claim: ', resultMessage)
-    throw new Error('Could not claim funds: ', resultMessage)
+const DEFAULT_CHANNEL_AMOUNT = 1000000
+const DEFAULT_BANDWIDTH = 2000
+const DEFAULT_FUND_THRESHOLD = 0.9
+
+class PluginXrpPaychan extends PluginBtp {
+  constructor (opts) {
+    super(opts)
+
+    this._xrpServer = opts.xrpServer || opts.rippledServer // TODO: deprecate rippledServer
+    this._api = new RippleAPI({ server: this._xrpServer })
+    this._secret = opts.secret
+    this._address = opts.address || deriveAddress(deriveKeypair(this._secret).publicKey)
+
+    this._peerAddress = opts.peerAddress // TODO: try to get this over the paychan?
+    this._bandwidth = opts.maxUnsecured || DEFAULT_BANDWIDTH
```

Comment:
`this._bandwidth` is not enforced. `_handleData` should check that the bandwidth is not exceeded before preparing a transfer.

Reply:
That's now connector functionality, so it doesn't belong in this plugin at all. I've taken that line out.